src/sas/sascalc/fit/BumpsFitting.py

```diff
@@ -297,17 +297,14 @@ def fit(self, msg_q=None,
                 param.value = uncertainties.ufloat(val, err)
         else:
             try:
-                uncertainties.correlated_values(values, cov)
-            except:
-                # No convergance 
-                for param, val, err in zip(varying, values, errs):
-                    # Convert all varying parameters to uncertainties objects
-                    param.value = uncertainties.ufloat(val, err)
-            else:
                 # Use the covariance matrix to calculate error in the parameter
                 fitted = uncertainties.correlated_values(values, cov)
                 for param, val in zip(varying, fitted):
                     param.value = val
+            except Exception:
+                # No convergence. Convert all varying parameters to uncertainties objects
+                for param, val, err in zip(varying, values, errs):
+                    param.value = uncertainties.ufloat(val, err)
```

Comment:
There is a small concern that the Exception code is eating an error unrelated to numerical stability. If for instance the sizes of `varying`, `values` and `cov` do not align. Since the except block will raise an exception on misaligned `varying` and `values`, it may only be `cov` that is messed up.

I propose an additional line **before** the try block (untested):
```
assert len(values) == cov.shape[0] == cov.shape[1]
```
If you had it in the try block then a failure would be trapped by the except block and you would still silently ignore it.

A couple of flags:

Do we really want to rely on python's `a cond b cond c` is equivalent to `a cond b and b cond c`? If you use `a == b == c` in most other languages you may be surprised by the result (I suspect it is undefined behaviour in many cases, depending on whether it is interpreted as `(a == b) == c` or `a == (b == c)` or just rejected as bad syntax).

IIRC, `assert` is suppressed if python is started with the optimization flag so we should not be relying on it for correct code.

You could be boring and use the following, but I prefer the compact assert statement:
```
if len(values) != cov.shape[0] or len(values) != cov.shape[1]:
    raise ValueError("Covariance matrix has the wrong shape")
```

Reply:
Fine. There was already an assert to check that values and errs existed, so I opted to add the simplest assert there as well.